src/TraceEvent/EventPipe/EventPipeEventSource.cs

```diff
@@ -83,6 +83,11 @@ private EventPipeEventSource(PinnedStreamReader streamReader, string name)
             StackCache = new StackCache();
         }
 
+        public DateTime QPCTimeToTimeStamp(long QPCTime)
```

Comment:
Question: Do we explicitly want a local time as opposed to a UTC time?

Also, either way, can you rename the API to specify the type of `DateTime` being returned (essentially using the name of the base API or if local time, perhaps `QPCTimeToLocalDateTime`)?

Reply:
I picked local since that is what you get from TraceEvent.TimeStamp and that is what you will compare with, I picked UTC first but when starting to compare with the TimeStamp of events you get a timezone diff you don't wan't. Didn't put it in the name to match TraceEvent.TimeStamp, its also a local DateTime.

Reply:
Gotcha.  Makes sense.  Can you add a test to ensure that we protect this code path?  It would be reasonable to check in a small nettrace next to the other files and use it to ensure that the new scenario that your dotnet/runtime PR enables works properly here.

Reply:
@brianrob Added test with minimal nettrace file only including 3 ExecutionCheckpoint events. Test uses timestamps in those events, convert to TimeStamp using new API and make sure they align with a set of predefined deltas.